NewForce-at-Mountwest/bangazonapi-team-crocodoom
# Perry product resource

## Description
This is a build of the Product Controller, as well as the Product Test Controller. The Developer should be able to use the Bangazon API to GET all Products, GET an individual Product by its Id, POST a new Product to the Database, update a single Product using PUT, and DELETE a Product. The Developer should also be able to Test Each Method previously mentioned through the Product Test Controller found in the BangazonAPITest. 
## Proposed changes
- This is a new feature which adds functionality for Products
-New Models were added for Product, Customer and Product Type as well as a Product Controller and a Product Test Controller
## Related Issues
A possible issue that could occur could be a Database conflict.
## Steps To Test
Before Testing, rebuild your database by running the SQL command listed in bangazon-api.sql. This should wipe the Database while also seeding it, making it consistent to the Database needed to run these Methods.

You will need to git checkout into perryProductResource branch .Cd into the BangazonAPI and use "dotnet run" to run the Program. Then open up a window of Postman in order to test each individual method. The url should be http://localhost:63540/api/Product to view all Products.

1.-GET - The first method to test is GET. This will get all of the products in the database. Copy the Url in the web browser and paste it in postman. Run the Get command and all of the Products should be shown.

2.-GET - The second method to test is GET a single product by Id.  Using the same url that was pasted into postman from the previous step, add a /0 to choose the first Product in the database.

3.-POST- The third method to test is POST. This will allow the creating of a new Product in the database. Using the original url from step 1, change the method in Postman to POST. Add a product using the JSON format with a ProductTypeId, CustomerId, Price, Title, Description, and Quantity. It should show your new Product when you GET using the same url.

4.-PUT- The next method to test is PUT. This will update a current Product in your database. Using the url from step 2 along with the /0, use GET to show the first Product. Change the method to PUT and edit the Title to "Test". Execute the PUT method then use GET to view your updated Product

5.-DELETE- The Last method to test is DELETE. This will Delete a current Product in the Database. Using the url from steps 2 and 4, use the GET to show the first Product. Change the method to DELETE and delete the Product.  Using the GET method, see the list of Products without the one deleted.

6- Go to View in the toolbar of Visual Studio, select Test Explorer. We are going to test the Methods by creating dummy data in the productTestController. In the top left of the Test explorer, click the green play button to run the Testing tasks. The results for each task should be a green arrow. 

## What kind of change does this PR introduce?
Please check one of the following:
- [x ] New feature

##Edits
-Included the PK in the GET method on ProductController.cs. This allows for the Developer to pull the correct Product by Id rather than defaulting to the PK of 0.
-Minor change to the PUT method where the Id is being pulled from the Route rather than the Body.
